pkg/services/replicator/process.go

```diff
@@ -43,7 +43,8 @@ func (p *Replicator) Run(ctx context.Context) {
 }
 
 // HandleTask executes replication task inside invoking goroutine.
-func (p *Replicator) HandleTask(ctx context.Context, task *Task) {
+// Returns slice of node IDs that has accepted the replica.
+func (p *Replicator) HandleTask(ctx context.Context, task *Task) []uint64 {
```

Comment:
thought about that

the only reason why i did so: that func is also [used](https://github.com/nspcc-dev/neofs-node/blob/1708cc0cc7aee468339027b5fb85d81e4ec199da/pkg/services/replicator/process.go#L40) that is never called in fact (cause [`AddTask`](https://github.com/nspcc-dev/neofs-node/blob/1708cc0cc7aee468339027b5fb85d81e4ec199da/pkg/services/replicator/task.go#L20) is not used). in that context, it is not clear what `checkedNodes` is

so i can pass `map` as a parameter and consider that it could be `nil` or just remove unused code. that do you think?

/cc @cthulhu-rider

Reply:
why is changing replicator bad?

without the replicator changes, it becomes extremely hard to implement any caching logic: as a policer we are only able to say "we have tried to replicate to `N` nodes `m` times" and have absolutely no feedback about what exactly nodes do have the replica now. it becomes impossible to prevent double replication in some cases

Reply:
Oh, I thought that `Replicator` solves task asynchronously (it does like that some time ago). Now it's synchronous, so your intention is absolutely legitimate.

> cause [AddTask](https://github.com/nspcc-dev/neofs-node/blob/1708cc0cc7aee468339027b5fb85d81e4ec199da/pkg/services/replicator/task.go#L20) is not used

Lets remove `AddTask` and accept success cache (btw we've always dodge this approach, and now it seems good :laughing:)
```
type TaskResult interface {
  SubmitSuccessfulNode(id uint64)
}

func (p *Replicator) HandleTask(ctx context.Context, task *Task, res TaskResult)
```
IMO interface is better in terms of abstraction and maintainability, but `map` is also OK.

Reply:
> Lets remove AddTask

removed in a separate commit

> IMO interface is better

added interface that is implemented by the type over `map`